libraries/protocol/include/steem/protocol/steem_required_actions.hpp

```diff
@@ -16,8 +18,76 @@ namespace steem { namespace protocol {
    };
 #endif
 
+#ifdef STEEM_ENABLE_SMT
+   struct smt_refund_action : public base_operation
+   {
+      asset_symbol_type symbol;
+      account_name_type contributor;
+      uint32_t          contribution_id;
+      asset             refund;
+
+      void validate()const;
+
+      friend bool operator==( const smt_refund_action& lhs, const smt_refund_action& rhs );
+   };
+
+   struct smt_contributor_payout_action : public base_operation
+   {
+      asset_symbol_type    symbol;
+      account_name_type    contributor;
+      uint32_t             contribution_id;
+      std::vector< asset > payouts;
+
+      void validate()const;
+
+      friend bool operator==( const smt_contributor_payout_action& lhs, const smt_contributor_payout_action& rhs );
+   };
+
+   struct smt_founder_payout_action : public base_operation
+   {
+      asset_symbol_type                                   symbol;
+      std::map< account_name_type, std::vector< asset > > payouts;
+
+      void validate()const;
+
+      friend bool operator==( const smt_founder_payout_action& lhs, const smt_founder_payout_action& rhs );
+   };
+
+   struct smt_ico_launch_action : public base_operation
```

Comment:
Add the control account. It is not needed for the action, but we need to statelessly get the control account so this can go in the account history.